visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -67,7 +66,55 @@ export class CodeChartaService {
 	private addFile(file: NameDataPair) {
 		const ccFile = getCCFile(file)
 		NodeDecorator.decorateMapWithPathAttribute(ccFile)
+		const currentFileChecksum = ccFile.fileMeta.fileChecksum
+		let currentFileName = ccFile.fileMeta.fileName
+		const storedFileNames = new Map(this.fileStates.map(file => [file.file.fileMeta.fileName, file.file.fileMeta.fileChecksum]))
+		const storedFileChecksums = new Map(
+			this.fileStates.map((file, index) => [file.file.fileMeta.fileChecksum, index] as [string, number])
+		)
+		const isDuplicate = storedFileChecksums.has(currentFileChecksum)
+
+		if (storedFileNames.has(currentFileName)) {
+			currentFileName = this.findFileName(currentFileName, isDuplicate, storedFileNames, currentFileChecksum)
+			ccFile.fileMeta.fileName = currentFileName
+		}
+		if (isDuplicate) {
+			this.fileStates[storedFileChecksums.get(currentFileChecksum)].file.fileMeta.fileName = currentFileName
+			this.recentFiles[0] = currentFileName
+			this.recentFiles.push(currentFileName)
+			return
+		}
+
 		this.fileStates.push({ file: ccFile, selectedAs: FileSelectionState.None })
+		this.recentFiles.push(currentFileName)
+	}
+
+	private findFileName(currentFileName: string, isDuplicate: boolean, storedFileNames: Map<string, string>, currentFileChecksum: string) {
```

Comment:
I would rename this function. `find` indicates that not always a file name can be found.